### PR TITLE
Keep track of recently rejected transactions with a rolling bloom filter (cont'd)

### PR DESCRIPTION
**Pull request by laanwj**
Peter Todd is on holiday, so continued from #6452
Added a commit that moves recentRejects initialization to the top of InitBlockIndex to avoid not initializing it under some conditions.

### CODE REVIEW

**Comment by sipa** [comment]
ACK (tested as part of #6470).

File: src/main.cpp [hunk 1]
```diff
@@ -162,6 +162,29 @@ namespace {
...
+     * AcceptToMemoryPool. These are not rerequested until the chain tip
+     * changes, at which point the entire filter is reset. Protected by
+     * cs_main.
+     *
+     * Without this filter we'd be re-requesting txs from each of our peers,
+     * increasing bandwidth consumption considerably. For instance, with 100
+     * peers, half of which relay a tx we don't accept, that might be a 50x
+     * bandwidth increase. A flooding attacker attempting to roll-over the
+     * filter using minimum-sized, 60byte, transactions might manage to send
+     * 1000/sec if we have fast peers, so we pick 120,000 to give our peers a
```
Comment (dcousens) [inline src/main.cpp]: Pick `120,000` what?

File: src/main.cpp [hunk 2]
```diff
@@ -162,6 +162,29 @@ namespace {
...
+     * two minute window to send invs to us.
+     *
+     * Decreasing the false positive rate is fairly cheap, so we pick one in a
+     * million to make it highly unlikely for users to have issues with this
+     * filter.
```
Comment (dcousens) [inline src/main.cpp]: Decreasing the false positive rate [for a valid transaction?] is fairly cheap, so we pick one in [a] million [1/1e6,  what is this unit?] to make it highly unlikely for users [legitimate transactions?] to have issues with this [passing this?] filter.

**Comment by dcousens** [comment]
concept ACK

**Pull request merged**